### PR TITLE
docs: explain and note service inputs

### PR DESCRIPTION
**Pull request by powersj**
Give some details that a plugin is a service input and how it might act differently. The note about the interval does not apply to all service plugins, just like how --test could work with some. Let me know if you think of any other items to include in the template.

fixes: #12868

### CODE REVIEW

File: plugins/inputs/cassandra/README.md [hunk 1]
```diff
@@ -21,6 +21,17 @@ Documentation][1]
...
+
+This plugin is a service input. Normal plugins gather metrics determined by the
+interval setting. Service plugins start a service to listens and waits for
+metrics or events to occur. Service plugins have two key differences from
+normal plugins:
+
+1. The agent or plug-in specific interval setting may not apply
+2. The CLI options of `--test`, `--test-wait`, and `--once` may not produce
+   output for this plugin
+
```
Comment (srebhan) [inline plugins/inputs/cassandra/README.md:34]: Same here...

**Review by srebhan — CHANGES_REQUESTED** [review]
I think you somehow automatically determine if a plugin is a service input by _only_ looking at the `Start()` / `Stop()` functions. However, this does not determine a service plugin! A service plugin generates metric **independent** of `Gather()` (i.e. that function is usually empty) which is not the case for quite a considerable number of the plugins in your list...

Do you know, I wanted to split those out to a separate directory a while ago... ;-P

**Comment by powersj** [comment]
@srebhan, 

> I think you somehow automatically determine if a plugin is a service input by only looking at the Start() / Stop() functions.

That is exactly what I did :smile: I decided to do that based off of the [`ServiceInput` interface](https://github.com/influxdata/telegraf/blob/master/input.go#L11).

> A service plugin generates metric independent of Gather()

Other than the two you called out, do I need to go through them again and verify?

> Do you know, I wanted to split those out to a separate directory a while ago... ;-P

I went looking for this issue and could not find it! :( I thought you had a list of them and was going to steal it from this.

Thanks for taking a look!

File: plugins/inputs/intel_pmu/README.md
Comment (Hipska) [inline plugins/inputs/intel_pmu/README.md:1]: NA

File: plugins/inputs/ethtool/README.md
Comment (Hipska) [inline plugins/inputs/ethtool/README.md:1]: NA

Not even implementing ServiceInput at all..

File: plugins/inputs/docker_log/README.md
Comment (Hipska) [inline plugins/inputs/docker_log/README.md:1]: NA

File: plugins/inputs/directory_monitor/README.md
Comment (Hipska) [inline plugins/inputs/directory_monitor/README.md:1]: NA

File: plugins/inputs/clickhouse/README.md
Comment (Hipska) [inline plugins/inputs/clickhouse/README.md:1]: NA

File: plugins/inputs/cassandra/README.md
Comment (Hipska) [inline plugins/inputs/cassandra/README.md:1]: NA

File: plugins/inputs/aliyuncms/README.md
Comment (Hipska) [inline plugins/inputs/aliyuncms/README.md:1]: NA

File: docs/includes/service_input.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+This plugin is a service input. Normal plugins gather metrics determined by the
+interval setting. Service plugins start a service to listens and waits for
+metrics or events to occur. Service plugins have two key differences from
+normal plugins:
+
+1. The agent or plug-in specific interval setting may not apply
```
Comment (Hipska) [inline docs/includes/service_input.md:6]: Make the naming more consistent with "Global configuration options" section.
```suggestion
1. The global or plugin specific `interval` setting may not apply
```

**Comment by srebhan** [comment]
@powersj see @Hipska's comments. Actually I think you need to go through the list as there is no easy machine-readable indication of a plugin being a service-plugin.

File: plugins/inputs/win_eventlog/README.md [hunk 3]
```diff
@@ -20,14 +20,16 @@ See the [CONFIGURATION.md][CONFIGURATION.md] for more details.
 
 [CONFIGURATION.md]: ../../../docs/CONFIGURATION.md#plugins
 
+## Service Input <!-- @/docs/includes/service_input.md -->
+
 ## Configuration
```
[the remark this answers is not in the document]
Comment (powersj) [reply plugins/inputs/win_eventlog/README.md:25]: yeah I fixed this in 9256a3f2e63045666a58b61d27660b9efdd6c437, I'll rebase to grab it

File: plugins/inputs/ethtool/README.md
Comment (powersj) [reply plugins/inputs/ethtool/README.md:1]: lol I think the `start` in `namespace_linux.go` got me

File: docs/includes/service_input.md (hunk 2 above)
Comment (powersj) [reply docs/includes/service_input.md:6]: thx applied

File: plugins/inputs/mqtt_consumer/README.md
[the remark this answers is not in the document]
Comment (powersj) [reply plugins/inputs/mqtt_consumer/README.md:1]: Gather is only used to ensure we are still connected + consumer name means we wait and listen for metrics -> I'm including this one as a service input

File: plugins/inputs/nsq_consumer/README.md
[the remark this answers is not in the document]
Comment (powersj) [reply plugins/inputs/nsq_consumer/README.md:1]: Another consumer to wait to receive metrics, so I'll include this as a service input

**Comment by powersj** [comment]
Thanks for the reviews and making it easy to update! I have applied the suggested changes: namely wording on the common text + removed it from input plugins that gather is actually used to get metrics.

> Furthermore it would make more sense for me to add this new section after the existing "Global configuration options" section.

I debated this and decided against it because this isn't really a config item but rather a high-level plugin note that way the flow is the following:

1. service plugin
2. global config options
3. config

What do you think?

**Review by Hipska — APPROVED** [review]

File: plugins/inputs/win_eventlog/README.md [hunk 4]
```diff
@@ -11,6 +11,17 @@ Windows Events Channels, like System Log.
 
 Telegraf minimum version: Telegraf 1.16.0
 
+## Service Input <!-- @/docs/includes/service_input.md -->
```
Comment (srebhan) [inline plugins/inputs/win_eventlog/README.md:14]: I think this one is not a service plugin.

**Review by srebhan — CHANGES_REQUESTED** [review]
Thanks for the update. I checked and `win_eventlog` is not a service input IMO. Can you please revert it?

File: plugins/inputs/win_eventlog/README.md (hunk 4 above)
Comment (powersj) [reply plugins/inputs/win_eventlog/README.md]: done

**Review by Hipska — APPROVED** [review]

**Pull request merged**